### DISCUSSION TITLE
React Navigation : native Stack Navigator / native-stack is much slower on Android since React Native 0.68.1

### DISCUSSION THREAD

**Discussion by tahola** [react-navigation/react-navigation#10591] [General]
Since I updated to Expo 45 and RN 0.68, @react-navigation/native-stack (StackActions.push() animation) is smooth but much slower that what it used to be on Android (I didn't notice any difference on IOS).

I also tried on a blank new app and the result is the same.

Is this part of the upgrade ? 

Is there a way to make it fast as before ?

https://user-images.githubusercontent.com/1329725/168463711-92323395-f952-415a-9483-3257295b2888.mp4

(It look ok on this video but its not, there is a delay, that doesnt feel like the native stack anymore)

**Comment by WoLewicki** [comment]
I think it is the result of this PR removing system animations on Android: https://github.com/software-mansion/react-native-screens/pull/1213. We changed it due to changes in `react-native` and we now make the animation by ourselves, so it might be slightly different unfortunately. The simplest way to change it for now would be to play with transition xmls: https://github.com/software-mansion/react-native-screens/blob/90e548739f35b5ded2524a9d6410033fc233f586/android/src/main/res/anim/rns_default_enter_in.xml etc.

**Reply by tahola** [reply]
Oh ok and can I change the transition speed there ? Can I modify that file `node_modules/react-native-screens/android/src/main/res/anim/rns_default_enter_in.xml` ? What should I do for the changes to take effect ? Thank you very much

**Reply by WoLewicki** [reply]
There are 4 transition xmls there, after changing the values in them, it should be enough to just build the app again and the changes should be visible then.

**Reply by hirbod** [reply]
And remember to use `patch-package` after that - or your changes will be lost after any npm/yarn update.

**Reply by tahola** [reply]
Thank you.

Not really related but while I am looking at it, testing and you guys seem to know a lot about it, I am wondering, it there a reason why the slide from right (rns_slide_in_from_right) animation is different than the slide from bottom (rns_slide_in_from_bottom) animation ? The slide from bottom slide from over the screen when the slide from right is pushing the screen to the left (less smoother...), is there a way to have it acting the same way ?

**Reply by WoLewicki** [reply]
Do you mean that with slide from right the screen below does not stay in one place but rather slides away too? To be honest, I cannot remember now why it was implemented in such way, but I believe you can change it in `xmls` so the below screen has no transition.

**Reply by jslok** [reply]
After the PR, it seems the library does not respect custom transition animation (.5x or 0) set by the user in developer options.

**Reply by zfy171** [reply]
> 我认为这是此 PR 删除 Android 上的系统动画的结果：[software-mansion/react-native-screens#1213](https://github.com/software-mansion/react-native-screens/pull/1213)。由于变化，`react-native`我们对其进行了更改，现在我们自己制作动画，因此不幸的是它可能会略有不同。目前更改它的最简单方法是使用过渡 xml：[https://github.com/software-mansion/react-native-screens/blob/90e548739f35b5ded2524a9d6410033fc233f586/android/src/main/res/anim/rns_default_enter_in。 xml](https://github.com/software-mansion/react-native-screens/blob/90e548739f35b5ded2524a9d6410033fc233f586/android/src/main/res/anim/rns_default_enter_in.xml)等

修改之后没有效果

**Reply by FusionReaction** [reply]
Wondering if anyone figured out a workaround, I tried updating the rns_default_enter_in.xml file but it didn't seem to make a difference. Any tips on what the ideal values would be? best.
